core/includes/updates/200-pr12.php

```diff
@@ -157,6 +157,49 @@
     // Continue
 }
 
+// delete language cache since it will contain the old language names and not the short codes
+$cache->setCache('languagecache');
+$cache->eraseAll();
+
+$default_language = $queries->getWhere('languages', ['is_default', '=', 1])[0]->name;
+
+// drop all & truncate from languages table
+try {
+    DB::getInstance()->createQuery('TRUNCATE TABLE nl2_languages');
+} catch (Exception $e) {
+    // Continue
+}
+
+// reset all user languages to default
+try {
+    DB::getInstance()->createQuery('UPDATE nl2_users SET `language_id` = ?', [
+        $default_language[0]->id
+    ]);
+} catch (Exception $e) {
+    // Continue
+}
+
+
+// add short code column to languages table
+try {
+    DB::getInstance()->createQuery('ALTER TABLE nl2_languages ADD `short_code` VARCHAR(64) NOT NULL');
+} catch (Exception $e) {
+    // Continue
+}
+
+// insert short codes into languages table, keeping the default language
+try {
+    foreach (Language::LANGUAGES as $short_code => $meta) {
+        DB::getInstance()->createQuery('INSERT INTO nl2_languages (`name`, `short_code`, `is_default`) VALUES (?, ?, ?)', [
+            $meta['name'],
+            $short_code,
+            $default_language === str_replace(' ', '', $meta['name']) ? 1 : 0
```

Comment:
If the old default language does not exist anymore (ie custom language), we should choose one of the languages to set as the new default - otherwise we're never going to have a default language

Reply:
I guess after this query we can check if any language exists where `is_default = 1`, and set ID = 1 to is_default, following similar logic as your comment previously?